src/components/ui/productgrid.tsx

```diff
@@ -0,0 +1,135 @@
+import { Button } from "@/components/ui/button"
+
+export interface Product {
+  id: number;
+  name: string;
+  price: number;
+  // add any additional properties as needed
+}
+
+const allProducts = [
+  {
+    id: 1,
+    name: "Organic Apples",
+    price: 3.99,
+    image: "/placeholder.svg?height=200&width=200",
+    vendor: "Green Valley Farm",
+    category: "Fruits",
+  },
+  {
+    id: 2,
+    name: "Fresh Carrots",
+    price: 2.49,
+    image: "/placeholder.svg?height=200&width=200",
+    vendor: "Organic Harvest Co.",
+    category: "Vegetables",
+  },
+  {
+    id: 3,
+    name: "Whole Milk",
+    price: 3.99,
+    image: "/placeholder.svg?height=200&width=200",
+    vendor: "Sunrise Farms",
+    category: "Dairy",
+  },
+  {
+    id: 4,
+    name: "Organic Beans",
+    price: 4.99,
+    image: "/placeholder.svg?height=200&width=200",
+    vendor: "Local Farmer Limited",
+    category: "Beans",
+  },
+  {
+    id: 5,
+    name: "Brown Rice",
+    price: 5.99,
+    image: "/placeholder.svg?height=200&width=200",
+    vendor: "Fresh Fields Co-op",
+    category: "Grains",
+  },
+  {
+    id: 6,
+    name: "Fresh Tomatoes",
+    price: 2.99,
+    image: "/placeholder.svg?height=200&width=200",
+    vendor: "Green Valley Farm",
+    category: "Vegetables",
+  },
+  {
+    id: 7,
+    name: "Organic Strawberries",
+    price: 4.99,
+    image: "/placeholder.svg?height=200&width=200",
+    vendor: "Organic Harvest Co.",
+    category: "Fruits",
+  },
+  {
+    id: 8,
+    name: "Greek Yogurt",
+    price: 3.49,
+    image: "/placeholder.svg?height=200&width=200",
+    vendor: "Sunrise Farms",
+    category: "Dairy",
+  },
+  {
+    id: 9,
+    name: "Quinoa",
+    price: 6.99,
+    image: "/placeholder.svg?height=200&width=200",
+    vendor: "Fresh Fields Co-op",
+    category: "Grains",
+  },
+]
+
+interface Filters {
+  categories: string[];
+  vendors: string[];
+  priceRange: [number, number];
+}
+
+export interface ProductGridProps {
+  filters: Filters;
+  sortBy: string;
+  onAddToCart: (product: Product) => void;
+}
+
+export function ProductGrid({ filters, sortBy, onAddToCart }: ProductGridProps) {
+  const filteredProducts = allProducts.filter((product) => {
```

Comment:
@Vivekt719 You won't be needing to do all the filtering yourself when you connect to backend